sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -39,22 +37,30 @@ export class PartitionPump {
   async start(): Promise<void> {
     this._isReceiving = true;
     try {
-      await this._partitionProcessor.initialize();
+      const requestedDefaultPosition = await this._partitionProcessor.initialize();
+
+      if (isEarliestEventPosition(this._initialEventPosition) && requestedDefaultPosition) {
```

Comment:
Can you add a comment that indicates `isEarliestEventPosition` is checking if `_initialEventPosition` is its default value? I know I'll forget as soon as we have a holiday!

Reply:
Your point is a good one. I ended up removing the "defaulting" and just having undefined be an actual value you can pass to indicate that you want the default.

So now it's explicit and should read a bit better.